Document/0x04c-Tampering-and-Reverse-Engineering.md

```diff
@@ -196,31 +197,40 @@ Even though most professional GUI-based disassemblers feature scripting faciliti
 
 In the Android section, you'll find a walkthrough for cracking a simple license check in an Android application using symbolic execution.
 
 ### References
 
-- [#vadla] Ole André Vadla Ravnås, Anatomy of a code tracer - <https://medium.com/@oleavr/anatomy-of-a-code-tracer-b081aadb0df8>
+- [#vadla] Ole André Vadla Ravnås, [Anatomy of a code tracer](https://medium.com/@oleavr/anatomy-of-a-code-tracer-b081aadb0df8 "Anatomy of a code tracer")
 
 #### OWASP Mobile Top 10 2016
 
-- M9 - Reverse Engineering - <https://www.owasp.org/index.php/Mobile_Top_10_2016-M9-Reverse_Engineering>
+- [M9 - Reverse Engineering](https://www.owasp.org/index.php/Mobile_Top_10_2016-M9-Reverse_Engineering "M9 - Reverse Engineering")
 
 #### Tools
 
-- Angr - <https://github.com/angr/angr>
-- Cycript - <http://www.cycript.org/>
-- Frida - <https://www.frida.re/>
-- Frida CLI - <https://www.frida.re/docs/frida-cli/>
-- frida-ls-devices - <https://www.frida.re/docs/frida-ls-devices/>
-- frida-ps - <https://www.frida.re/docs/frida-ps/>
-- frida-trace - <https://www.frida.re/docs/frida-trace/>
-- Fridump - <https://github.com/Nightbringer21/fridump>
-- Objection - <https://github.com/sensepost/objection>
-- Passionfruit - <https://github.com/chaitin/passionfruit>
-- r2frida - <https://github.com/nowsecure/r2frida>
-- Radare2 - <https://github.com/radare/radare2>
-- Substrate - <http://www.cydiasubstrate.com/>
-- Xposed - <https://www.xda-developers.com/xposed-framework-hub/>
+- [Angr](https://github.com/angr/angr "Angr")
+- [Cycript](http://www.cycript.org/ "Cycript")
+- [Frida](https://www.frida.re/ "Frida")
+- [Frida CLI](https://www.frida.re/docs/frida-cli/ "Frida CLI")
+- [frida-ls-devices](https://www.frida.re/docs/frida-ls-devices/ "frida-ls-devices")
+- [frida-ps](https://www.frida.re/docs/frida-ps/ "frida-ps")
+- [frida-trace](https://www.frida.re/docs/frida-trace/ "frida-trace")
+- [Fridump](https://github.com/Nightbringer21/fridump "Fridump")
+- [Objection](https://github.com/sensepost/objection "Objection")
+- [Passionfruit](https://github.com/chaitin/passionfruit "Passionfruit")
+- [r2frida](https://github.com/nowsecure/r2frida "r2frida")
+- [Radare2](https://github.com/radare/radare2 "Radare2")
+- [Substrate](http://www.cydiasubstrate.com/ "Substrate")
+- [Xposed](https://www.xda-developers.com/xposed-framework-hub/ "Xposed")
```

Comment:
@cpholguera / @su-vikas / @TheDauntless : this will be harder to follow when you read the book. What if all tools would be formatted as: `[Toolname - url](URL, "toolname")`?

Reply:
I agree with your point on readability, but not sure on how to fix it.

Reply:
Good thinking!

@su-vikas  You can do this by adding the URL inside the first part:

```
[Substrate - http://www.cydiasubstrate.com/](http://www.cydiasubstrate.com/ "Substrate")
```
That way you can see the URL in the book, and click it online.

Reply:
@commjoen @TheDauntless we've agreed some time ago on a format for the references links. That is `- name - <url>`, which is also printable book friendly. e.g.:

`- adb - <https://developer.android.com/studio/command-line/adb>`

It took us long to implement overall in the guide so that it is consistent. We should avoid changing this format once more as it is already what we wanted :)

@su-vikas the fix here would be to revert those changes on the references and apply the same style to your added references. Thank you very much!

Reply:
I have reverted the changes.